python/lsst/analysis/tools/tasks/limitingSurfBriAnalysis.py

```diff
@@ -0,0 +1,338 @@
+# This file is part of analysis_tools.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = (
+    "LimitingSurfBriTask",
+    "LimitingSurfBriConfig",
+    "LimitingSurfBriConnections",
+    "LimitingSurfBriAnalysisConnections",
+    "LimitingSurfBriAnalysisConfig",
+    "LimitingSurfBriAnalysisTask",
+)
+
+import logging
+
+import lsst.afw.math as afwMath
+import numpy as np
+from astropy.table import Table
+from lsst.pex.config import Field, ListField
+from lsst.pipe.base import PipelineTask, PipelineTaskConfig, PipelineTaskConnections, Struct
+from lsst.pipe.base.connectionTypes import Input, Output
+from lsst.skymap import BaseSkyMap
+
+from ..interfaces import AnalysisBaseConfig, AnalysisBaseConnections, AnalysisPipelineTask
+
+log = logging.getLogger(__name__)
+
+
+class LimitingSurfBriConnections(
+    PipelineTaskConnections,
+    dimensions=(),
+    defaultTemplates={
+        "detectionTableName": "",
+        "photoCalibName": "",
+        "wcsName": "",
+    },
+):
+    """Connections class for LimitingSurfBriTask."""
+
+    data = Input(
+        doc="Visit- or coadd-level object table",
+        name="{detectionTableName}",
+        storageClass="ArrowAstropy",
+        multiple=True,
+        dimensions=(),
```

Comment:
I don't think all of these dimensions should be empty.

Reply:
I think (@aemerywatkins can correct me if wrong) that these have been left empty on purpose because the dimensions are set programatically by the user in the, e.g., `inputTableDimensions`. 

Leaving this blank may help make it clear that these are set elsewhere? We could populate them, but they'll never be used, so it would be source-code-only code, if that has any use at all?